docs/api/utilities.md

```diff
@@ -0,0 +1,412 @@
+  console.log('LLM switched:', data.newConfig);
+});
+
+sessionBus.on('llmservice:thinking', () => {
+  console.log('LLM is thinking...');
+});
+
+// Emit events
+agentBus.emit('saiki:conversationReset', { sessionId: 'session-1' });
+```
+
+---
+
+## Storage Factory
+
+### `createStorageBackends`
+
+Creates storage backend instances for cache and database operations.
+
+```typescript
+async createStorageBackends(config: StorageConfig): Promise<{
+  manager: StorageManager;
+  backends: StorageBackends;
+}>
+```
+
+### Storage Backend Interfaces
+
+#### Cache Backend
+
+```typescript
+interface CacheBackend {
+  get(key: string): Promise<any>;
+  set(key: string, value: any, ttl?: number): Promise<void>;
+  delete(key: string): Promise<void>;
+  clear(): Promise<void>;
+}
+```
+
+#### Database Backend
+
+```typescript
+interface DatabaseBackend {
+  get(key: string): Promise<any>;
+  set(key: string, value: any): Promise<void>;
+  delete(key: string): Promise<void>;
+  append(key: string, value: any): Promise<void>;
+  getRange(key: string, start: number, end: number): Promise<any[]>;
+}
+```
+
+### Example Usage
+
+```typescript
+import { createStorageBackends } from '@truffle-ai/saiki';
+
+const { manager, backends } = await createStorageBackends({
+  cache: { type: 'in-memory' },
+  database: { type: 'in-memory' }
+});
+
+// Use cache for temporary data
+await backends.cache.set('session:123', sessionData, 3600); // 1 hour TTL
+const sessionData = await backends.cache.get('session:123');
+
+// Use database for persistent data
+await backends.database.set('user:456', userData);
+await backends.database.append('messages:789', message);
+const messages = await backends.database.getRange('messages:789', 0, 50);
+
+// Cleanup when done
+await manager.disconnect();
+```
+
+---
+
+## Service Factory
+
+### `createAgentServices`
+
+Low-level factory for creating all agent services with proper dependency injection.
+
+```typescript
+async createAgentServices(
+  agentConfig: AgentConfig,
+  cliArgs?: CLIConfigOverrides,
+  overrides?: InitializeServicesOptions
+): Promise<AgentServices>
+```
+
+**Parameters:**
+- `agentConfig` - Complete agent configuration
+- `cliArgs` - Optional CLI argument overrides
+- `overrides` - Optional service instance overrides
+
+**Returns:** `AgentServices` object containing:
+- `clientManager: MCPManager`
+- `promptManager: PromptManager`
+- `agentEventBus: AgentEventBus`
+- `stateManager: AgentStateManager`
+- `sessionManager: SessionManager`
+- `storage: StorageBackends`
+
+### Example Usage
+
+```typescript
+import { createAgentServices } from '@truffle-ai/saiki';
+
+const services = await createAgentServices(config, overrides, options);
+
+// Access individual services
+const { clientManager, sessionManager, stateManager } = services;
+
+// Use services independently
+const tools = await clientManager.getAllTools();
+const session = await sessionManager.createSession();
+``` 
```

Comment:
imo we don't need any of this. logger and event bus docs we can take out for sure.

maybe `createAgentServices` might be useful but i don't think we should push people to use the library this way.
let people work with `SaikiAgent` and `MCPManager`

storage docs are already there also so we don't need that here either